ChallengeTrackerApplication/ChallengeTrackerApplication-ejb/src/java/ejb/session/stateless/UserEntitySessionBean.java

```diff
@@ -23,6 +35,41 @@
     @PersistenceContext(unitName = "ChallengeTrackerApplication-ejbPU")
     private EntityManager em;
     
+    private final ValidatorFactory validatorFactory;
+    private final Validator validator;
+
+    public UserEntitySessionBean() {
+        validatorFactory = Validation.buildDefaultValidatorFactory();
+        validator = validatorFactory.getValidator();
+    }
+
+
+    @Override
+    public Long createNewUser(User newUserEntity) throws UserUsernameExistException, UnknownPersistenceException, InputDataValidationException {
+        Set<ConstraintViolation<User>> constraintViolations = validator.validate(newUserEntity);
+
+        if (constraintViolations.isEmpty()) {
+            try {
+
+                em.persist(newUserEntity);
+                em.flush();
+
+                return newUserEntity.getUserId();
+            } catch (PersistenceException ex) {
+                if (ex.getCause() != null && ex.getCause().getClass().getName().equals("org.eclipse.persistence.exceptions.DatabaseException")) {
+                    if (ex.getCause().getCause() != null && ex.getCause().getCause().getClass().getName().equals("java.sql.SQLIntegrityConstraintViolationException")) {
```

Comment:
so many if statements, please refactor to make intent is clear and for readability

Reply:
done